### PR TITLE
[BPK-903] Add token switcher

### PR DESCRIPTION
**Pull request by matthewdavidson**
+ [x] For any new web components I've made sure that the style can be extended by the consumer using a `className` override.
+ [x] For any new native components I've made sure that the style can be extended by the consumer using a `style` override.
+ [x] For any new files I've included the [Apache 2.0 License header](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-tokens/formatters/license-header.js#L2-L16) at the top of the file.
+ [x] I've updated `changelog.md` for any changes that need to be highlighted in the changelog.
+ [x] If I've updated `npm-shrinkwrap.json` those changes are intentional.

### CODE REVIEW

File: packages/bpk-docs/src/components/DocsPageBuilder/TokenSwitcher.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  },
+  android: {
+    id: 'android',
+    name: 'Android',
+  },
+};
+
+class TokenSwitcher extends Component {
+  constructor() {
+    super();
```
Comment (matteo-hertel) [inline packages/bpk-docs/src/components/DocsPageBuilder/TokenSwitcher.js]: Just a minor thing, but we should accept props in the constructor signature and pass it to `super`

```
constructor(props) {
  super(props);
```

File: packages/bpk-docs/src/pages/ZIndexesPage/ZIndexesPage.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import TOKENS from 'bpk-tokens/tokens/base.raw.json';
+import IOS_TOKENS from 'bpk-tokens/tokens/ios/base.raw.json';
+import ANDROID_TOKENS from 'bpk-tokens/tokens/android/base.raw.json';
+
+import DocsPageBuilder from './../../components/DocsPageBuilder';
+import { getPlatformTokens } from './../../helpers/tokens-helper';
+
+const ZIndexesPage = () => <DocsPageBuilder
+  title="Z-Indexes"
+  tokenMap={getPlatformTokens(TOKENS, IOS_TOKENS, ANDROID_TOKENS, ({ category }) => category === 'z-indices')}
```
Comment (matteo-hertel) [inline packages/bpk-docs/src/pages/ZIndexesPage/ZIndexesPage.js]: @shaundon indices or indexes? 😄

Comment (matthewdavidson) [reply packages/bpk-docs/src/pages/ZIndexesPage/ZIndexesPage.js]: Ha so I stumbled across this http://grammarist.com/usage/indexes-indices/.

Indexes sounds better - nobody says indices 😂 .

However we cannot change the token category which unfortunately is called indices.

Comment (shaundon) [reply packages/bpk-docs/src/pages/ZIndexesPage/ZIndexesPage.js]: Yeah I just googled it and apparently both are good :P

File: packages/bpk-docs/src/components/DocsPageBuilder/TokenSwitcher.js (hunk 1 above)
Comment (matthewdavidson) [reply packages/bpk-docs/src/components/DocsPageBuilder/TokenSwitcher.js]: Technically you only need to pass props to super if you intend to access `this.props` in the constructor.

However, I reckon you probably just want us to be consistent, in that case I reckon we should make that change across the board in a separate PR - currently we only pass props to super when required.

Comment (matteo-hertel) [reply packages/bpk-docs/src/components/DocsPageBuilder/TokenSwitcher.js]: Make sense, cheers

**Pull request merged**